File: 012_Integer_To_Roman/solution.go
func intToRoman(num int) string {
    symbol := [...]string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}
    value := [...]int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
    str := ""
    temp := num
    for i := 0; temp != 0; i ++ {
        for temp >= value[i] {
            temp -= value[i]
            str += symbol[i]
        }
    }
    return str
}

File: 012_Integer_To_Roman/solution.py
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        syb = ['M', 'CM', 'D', 'CD', 
               'C', 'XC', 'L', 'XL',
               'X', 'IX', 'V', 'IV',
               'I']
        val = [1000, 900, 500, 400,
               100, 90, 50, 40,
               10, 9, 5, 4, 1]
        roma = ''
        i = 0
        while num > 0:
            for _ in range(num // val[i]):
                roma += syb[i]
                num -= val[i]
            i += 1
        return roma
